src/services/outliningElementsCollector.ts

```diff
@@ -3,12 +3,18 @@ namespace ts.OutliningElementsCollector {
     export function collectElements(sourceFile: SourceFile, cancellationToken: CancellationToken): OutliningSpan[] {
         const elements: OutliningSpan[] = [];
         const collapseText = "...";
+        let depth = 0;
+        const maxDepth = 20;
+
+        walk(sourceFile);
+        return elements;
 
-        function addOutliningSpan(hintSpanNode: Node, startElement: Node, endElement: Node, autoCollapse: boolean) {
+        // If useFullStart is true, then the collapsing span includes leading whitespace, including linebreaks.
+        function addOutliningSpan(hintSpanNode: Node, startElement: Node, endElement: Node, autoCollapse: boolean, useFullStart: boolean) {
             if (hintSpanNode && startElement && endElement) {
                 const span: OutliningSpan = {
-                    textSpan: createTextSpanFromBounds(startElement.pos, endElement.end),
-                    hintSpan: createTextSpanFromNode(hintSpanNode, sourceFile),
+                    textSpan: createTextSpanFromBounds(useFullStart ? startElement.pos : startElement.getStart(), endElement.end),

```

Comment:
I can't use `createTextSpanFromNode` everywhere because in most places, the start node and end node are different.
I'll change it back in the one relevant place though.

Also changing all .pos to .getFullStart (and similar for .end).